### ISSUE TITLE
BBC programme 'Ordeal by Innocence' Episode 2 doesn't download HD version

### ISSUE THREAD

**Issue by jefro108** [Ascoware/get-iplayer-automator#165]
The programme appears to be HD on the iPlayer website https://www.bbc.co.uk/iplayer/episode/b09zg9dd/ordeal-by-innocence-series-1-episode-2 but GiA downloads a 960x540 version. Get iPlayer Automator 1.13.2

```
BBC Download (ID=b09zg9dd): Downloading Ordeal by Innocence: Series 1 - Episode 2
INFO: encodinglocale = UTF-8
INFO: encodinglocalefs = UTF-8
INFO: encodingconsoleout = UTF-8
INFO: encodingconsolein = UTF-8
INFO: ${^UNICODE} = 39
INFO: Profile dir: /Users/user/Library/Application Support/Get iPlayer Automator
INFO: User options file: /Users/user/Library/Application Support/Get iPlayer Automator/options
INFO: System options file: /etc/get_iplayer/options
INFO: Search args: ''
INFO: Got 1353 file cache entries for itv
INFO: Got 7869 file cache entries for tv
INFO: Got 16852 file cache entries for radio
INFO: Cleaning PID - old: 'b09zg9dd' new: 'b09zg9dd'
INFO: Getting URL: http://www.bbc.co.uk/programmes/b09zg9dd.json
INFO: tv episode PID detected (b09zg9dd)
INFO: 1 total programmes
INFO: Loaded history for first check.
INFO: Loading recordings history
INFO: Programme not in history
INFO: Getting URL: http://www.bbc.co.uk/programmes/b09zg9dd.json
INFO: Getting URL: http://open.live.bbc.co.uk/mediaselector/5/select/version/2.0/mediaset/pc/vpid/p063prtg?cb=09089
INFO: Getting URL: http://vod-dash-uk-live.akamaized.net/usp/auth/vod/piff_abr_full_sd_ad/726a98-b0b0b6gz/vf_b0b0b6gz_fc477bc7-5e9d-467a-85cc-b8ab07192b95.ism/pc_hd_abr_v2_dash_master.mpd?__gda__=1523422637_8cf11679d776c72d9d395b96b8f93fe3
INFO: Getting URL: http://vod-dash-uk-live.bbcfmt.hs.llnwd.net/usp/auth/vod/piff_abr_full_sd_ad/726a98-b0b0b6gz/vf_b0b0b6gz_fc477bc7-5e9d-467a-85cc-b8ab07192b95.ism/pc_hd_abr_v2_dash_master.mpd?s=1523379437&e=1523422637&h=e0febfaa517ffe090cf97cc3f62ba55d
INFO: Getting URL: http://open.live.bbc.co.uk/mediaselector/5/select/version/2.0/mediaset/iptv-all/vpid/p063prtg?cb=31282
INFO: Getting URL: http://vod-hls-uk-live.akamaized.net/usp/auth/vod/piff_abr_full_sd_ad/726a98-b0b0b6gz/vf_b0b0b6gz_fc477bc7-5e9d-467a-85cc-b8ab07192b95.ism.hlsv2.ism/vf_b0b0b6gz_fc477bc7-5e9d-467a-85cc-b8ab07192b95.ism.hlsv2.m3u8?__gda__=1523422656_ce6ac8453b40f4a84382211865cd8cd2
INFO: Getting URL: http://vod-hls-uk-live.bbcfmt.hs.llnwd.net/usp/auth/vod/piff_abr_full_sd_ad/726a98-b0b0b6gz/vf_b0b0b6gz_fc477bc7-5e9d-467a-85cc-b8ab07192b95.ism.hlsv2.ism/vf_b0b0b6gz_fc477bc7-5e9d-467a-85cc-b8ab07192b95.ism.hlsv2.m3u8?s=1523379456&e=1523422656&h=c1c9007dfaa7433f8cc84a71a1eef637
INFO: Getting URL: http://mm.bidi.bbc.co.uk/vod-hls-uk-live/usp/auth/vod/piff_abr_full_sd_ad/726a98-b0b0b6gz/vf_b0b0b6gz_fc477bc7-5e9d-467a-85cc-b8ab07192b95.ism.hlsv2.ism/vf_b0b0b6gz_fc477bc7-5e9d-467a-85cc-b8ab07192b95.ism.hlsv2.m3u8?at=464ZvuoVd6b52a0b02803cd83b753d2609d08e1dd98ca6395698672cb1800
INFO: Found mode    dvfxsd1:   gip_dvf_iplayer_1604 dash h264   960x540 25fps 1604kbps mf_limelight_uk_dash/2
INFO: Found mode    dvfxsd2:   gip_dvf_iplayer_1604 dash h264   960x540 25fps 1604kbps mf_akamai_uk_dash/1
INFO: Found mode  dvfxhigh1:    gip_dvf_iplayer_827 dash h264   704x396 25fps  827kbps mf_limelight_uk_dash/2
INFO: Found mode  dvfxhigh2:    gip_dvf_iplayer_827 dash h264   704x396 25fps  827kbps mf_akamai_uk_dash/1
INFO: Found mode    dvflow1:    gip_dvf_iplayer_437 dash h264   512x288 25fps  437kbps mf_limelight_uk_dash/2
INFO: Found mode    dvflow2:    gip_dvf_iplayer_437 dash h264   512x288 25fps  437kbps mf_akamai_uk_dash/1
INFO: Found mode    hvfxsd1:   gip_hvf_iplayer_1836  hls h264   960x540 25fps 1836kbps mf_bidi_uk_hls/3
INFO: Found mode    hvfxsd2:   gip_hvf_iplayer_1836  hls h264   960x540 25fps 1836kbps mf_limelight_uk_hls/1
INFO: Found mode    hvfxsd3:   gip_hvf_iplayer_1836  hls h264   960x540 25fps 1836kbps mf_akamai_uk_hls/1
INFO: Found mode   hvfhigh1:   gip_hvf_iplayer_1800  hls h264   704x396 50fps 1800kbps mf_bidi_uk_hls/3
INFO: Found mode   hvfhigh2:   gip_hvf_iplayer_1800  hls h264   704x396 50fps 1800kbps mf_limelight_uk_hls/1
INFO: Found mode   hvfhigh3:   gip_hvf_iplayer_1800  hls h264   704x396 50fps 1800kbps mf_akamai_uk_hls/1
INFO: Found mode  hvfxhigh1:   gip_hvf_iplayer_1013  hls h264   704x396 25fps 1013kbps mf_bidi_uk_hls/3
INFO: Found mode  hvfxhigh2:   gip_hvf_iplayer_1013  hls h264   704x396 25fps 1013kbps mf_limelight_uk_hls/1
INFO: Found mode  hvfxhigh3:   gip_hvf_iplayer_1013  hls h264   704x396 25fps 1013kbps mf_akamai_uk_hls/1
INFO: Found mode    hvfstd1:    gip_hvf_iplayer_865  hls h264   640x360 25fps  865kbps mf_bidi_uk_hls/3
INFO: Found mode    hvfstd2:    gip_hvf_iplayer_865  hls h264   640x360 25fps  865kbps mf_limelight_uk_hls/1
INFO: Found mode    hvfstd3:    gip_hvf_iplayer_865  hls h264   640x360 25fps  865kbps mf_akamai_uk_hls/1
INFO: Found mode    hvflow1:    gip_hvf_iplayer_599  hls h264   512x288 25fps  599kbps mf_bidi_uk_hls/3
INFO: Found mode    hvflow2:    gip_hvf_iplayer_599  hls h264   512x288 25fps  599kbps mf_limelight_uk_hls/1
INFO: Found mode    hvflow3:    gip_hvf_iplayer_599  hls h264   512x288 25fps  599kbps mf_akamai_uk_hls/1
INFO: Getting URL: http://open.live.bbc.co.uk/mediaselector/5/select/version/2.0/mediaset/pc/vpid/b0b0b6gz?cb=05525
INFO: Getting URL: http://vod-dash-uk-live.akamaized.net/usp/auth/vod/piff_abr_full_hd/726a98-b0b0b6gz/vf_b0b0b6gz_7bb0e2d6-4be6-4b92-a31d-c3b0b685e513.ism/pc_hd_abr_v2_dash_master.mpd?__gda__=1523422657_89529acd6c21906a1cc98458aaa7c4eb
INFO: Getting URL: http://vod-dash-uk-live.bbcfmt.hs.llnwd.net/usp/auth/vod/piff_abr_full_hd/726a98-b0b0b6gz/vf_b0b0b6gz_7bb0e2d6-4be6-4b92-a31d-c3b0b685e513.ism/pc_hd_abr_v2_dash_master.mpd?s=1523379457&e=1523422657&h=b440fb62566a15fc45c00a106b8ea97e
INFO: Getting URL: http://open.live.bbc.co.uk/mediaselector/5/select/version/2.0/mediaset/iptv-all/vpid/b0b0b6gz?cb=66568
INFO: Getting URL: http://vod-hls-uk-live.akamaized.net/usp/auth/vod/piff_abr_full_hd/726a98-b0b0b6gz/vf_b0b0b6gz_7bb0e2d6-4be6-4b92-a31d-c3b0b685e513.ism.hlsv2.ism/vf_b0b0b6gz_7bb0e2d6-4be6-4b92-a31d-c3b0b685e513.ism.hlsv2.m3u8?__gda__=1523422430_28645d7476328fd5626cc5342702fdca
INFO: Getting URL: http://vod-hls-uk-live.bbcfmt.hs.llnwd.net/usp/auth/vod/piff_abr_full_hd/726a98-b0b0b6gz/vf_b0b0b6gz_7bb0e2d6-4be6-4b92-a31d-c3b0b685e513.ism.hlsv2.ism/vf_b0b0b6gz_7bb0e2d6-4be6-4b92-a31d-c3b0b685e513.ism.hlsv2.m3u8?s=1523379230&e=1523422430&h=ca3848a09a9ddc848b3e97dc9c0eb34c
INFO: Getting URL: http://mm.bidi.bbc.co.uk/vod-hls-uk-live/usp/auth/vod/piff_abr_full_hd/726a98-b0b0b6gz/vf_b0b0b6gz_7bb0e2d6-4be6-4b92-a31d-c3b0b685e513.ism.hlsv2.ism/vf_b0b0b6gz_7bb0e2d6-4be6-4b92-a31d-c3b0b685e513.ism.hlsv2.m3u8?at=IS4uIEXgf166bc6796cdf1ffdb838d5db4866c449799025d5698665529b80
INFO: Found mode     dvfhd1:   gip_dvf_iplayer_5070 dash h264  1280x720 50fps 5070kbps mf_limelight_uk_dash/2
INFO: Found mode     dvfhd2:   gip_dvf_iplayer_5070 dash h264  1280x720 50fps 5070kbps mf_akamai_uk_dash/1
INFO: Found mode     dvfsd1:   gip_dvf_iplayer_2812 dash h264   960x540 50fps 2812kbps mf_limelight_uk_dash/2
INFO: Found mode     dvfsd2:   gip_dvf_iplayer_2812 dash h264   960x540 50fps 2812kbps mf_akamai_uk_dash/1
INFO: Found mode    dvfxsd1:   gip_dvf_iplayer_1604 dash h264   960x540 25fps 1604kbps mf_limelight_uk_dash/2
INFO: Found mode    dvfxsd2:   gip_dvf_iplayer_1604 dash h264   960x540 25fps 1604kbps mf_akamai_uk_dash/1
INFO: Found mode  dvfxhigh1:    gip_dvf_iplayer_827 dash h264   704x396 25fps  827kbps mf_limelight_uk_dash/2
INFO: Found mode  dvfxhigh2:    gip_dvf_iplayer_827 dash h264   704x396 25fps  827kbps mf_akamai_uk_dash/1
INFO: Found mode    dvflow1:    gip_dvf_iplayer_437 dash h264   512x288 25fps  437kbps mf_limelight_uk_dash/2
INFO: Found mode    dvflow2:    gip_dvf_iplayer_437 dash h264   512x288 25fps  437kbps mf_akamai_uk_dash/1
INFO: Found mode     hvfhd1:   gip_hvf_iplayer_5714  hls h264  1280x720 50fps 5714kbps mf_bidi_uk_hls/3
INFO: Found mode     hvfhd2:   gip_hvf_iplayer_5714  hls h264  1280x720 50fps 5714kbps mf_limelight_uk_hls/2
INFO: Found mode     hvfhd3:   gip_hvf_iplayer_5714  hls h264  1280x720 50fps 5714kbps mf_akamai_uk_hls/1
INFO: Found mode     hvfsd1:   gip_hvf_iplayer_3320  hls h264   960x540 50fps 3320kbps mf_bidi_uk_hls/3
INFO: Found mode     hvfsd2:   gip_hvf_iplayer_3320  hls h264   960x540 50fps 3320kbps mf_limelight_uk_hls/2
INFO: Found mode     hvfsd3:   gip_hvf_iplayer_3320  hls h264   960x540 50fps 3320kbps mf_akamai_uk_hls/1
INFO: Found mode    hvfxsd1:   gip_hvf_iplayer_2040  hls h264   960x540 25fps 2040kbps mf_bidi_uk_hls/3
INFO: Found mode    hvfxsd2:   gip_hvf_iplayer_2040  hls h264   960x540 25fps 2040kbps mf_limelight_uk_hls/2
INFO: Found mode    hvfxsd3:   gip_hvf_iplayer_2040  hls h264   960x540 25fps 2040kbps mf_akamai_uk_hls/1
INFO: Found mode   hvfhigh1:   gip_hvf_iplayer_2004  hls h264   704x396 50fps 2004kbps mf_bidi_uk_hls/3
INFO: Found mode   hvfhigh2:   gip_hvf_iplayer_2004  hls h264   704x396 50fps 2004kbps mf_limelight_uk_hls/2
INFO: Found mode   hvfhigh3:   gip_hvf_iplayer_2004  hls h264   704x396 50fps 2004kbps mf_akamai_uk_hls/1
INFO: Found mode  hvfxhigh1:   gip_hvf_iplayer_1216  hls h264   704x396 25fps 1216kbps mf_bidi_uk_hls/3
INFO: Found mode  hvfxhigh2:   gip_hvf_iplayer_1216  hls h264   704x396 25fps 1216kbps mf_limelight_uk_hls/2
INFO: Found mode  hvfxhigh3:   gip_hvf_iplayer_1216  hls h264   704x396 25fps 1216kbps mf_akamai_uk_hls/1
INFO: Found mode    hvfstd1:   gip_hvf_iplayer_1069  hls h264   640x360 25fps 1069kbps mf_bidi_uk_hls/3
INFO: Found mode    hvfstd2:   gip_hvf_iplayer_1069  hls h264   640x360 25fps 1069kbps mf_limelight_uk_hls/2
INFO: Found mode    hvfstd3:   gip_hvf_iplayer_1069  hls h264   640x360 25fps 1069kbps mf_akamai_uk_hls/1
INFO: Found mode    hvflow1:    gip_hvf_iplayer_803  hls h264   512x288 25fps  803kbps mf_bidi_uk_hls/3
INFO: Found mode    hvflow2:    gip_hvf_iplayer_803  hls h264   512x288 25fps  803kbps mf_limelight_uk_hls/2
INFO: Found mode    hvflow3:    gip_hvf_iplayer_803  hls h264   512x288 25fps  803kbps mf_akamai_uk_hls/1
INFO: Found mode subtitles1:               captions http  N/A       N/A   N/A      N/A mf_limelight_uk_plain/20
INFO: Found mode subtitles2:               captions http  N/A       N/A   N/A      N/A sis/10
INFO: Found mode subtitles3:               captions http  N/A       N/A   N/A      N/A mf_akamai_uk_plain/10
INFO: Getting URL: http://open.live.bbc.co.uk/mediaselector/5/select/version/2.0/mediaset/pc/vpid/b09zg9bd?cb=04757
ERROR: Response: 404 Not Found
INFO: Getting URL: http://open.live.bbc.co.uk/mediaselector/5/select/version/2.0/mediaset/iptv-all/vpid/b09zg9bd?cb=89172
ERROR: Response: 404 Not Found
INFO: No streams available for 'original' version (b09zg9bd) - skipping
INFO: No streams found for 'original' version (b09zg9bd) - deleting
INFO: Programme not in history
INFO: Searching for versions: editorial
WARNING: A UK TV licence is required to access BBC iPlayer TV content legally
INFO: Mode list: radiobest,radiobetter,radiovgood,hlshd,hvfxsd,dvfxsd,hlsvhigh,hvfxhigh,dvfxhigh,hvfstd,hlsstd,hlsxstd,hvflow,dvflow,hvfxsd,dvfxsd,hlsvhigh,hvfxhigh,dvfxhigh,hvfstd,hlsstd,hlsxstd,hvflow,dvflow,hlsvhigh,hvfxhigh,dvfxhigh,hvfstd,hlsstd,hlsxstd,hvflow,dvflow
INFO: Searching for version: 'editorial'
INFO: Found version: 'editorial'
INFO: Modes to try for 'editorial' version: hvfxsd1,hvfxsd2,hvfxsd3,dvfxsd1,dvfxsd2,hvfxhigh1,hvfxhigh2,hvfxhigh3,dvfxhigh1,dvfxhigh2,hvfstd1,hvfstd2,hvfstd3,hvflow1,hvflow2,hvflow3,dvflow1,dvflow2,hvfxsd1,hvfxsd2,hvfxsd3,dvfxsd1,dvfxsd2,hvfxhigh1,hvfxhigh2,hvfxhigh3,dvfxhigh1,dvfxhigh2,hvfstd1,hvfstd2,hvfstd3,hvflow1,hvflow2,hvflow3,dvflow1,dvflow2,hvfxhigh1,hvfxhigh2,hvfxhigh3,dvfxhigh1,dvfxhigh2,hvfstd1,hvfstd2,hvfstd3,hvflow1,hvflow2,hvflow3,dvflow1,dvflow2
INFO: Downloading tv: 'Ordeal by Innocence: Series 1 - 2. Episode 2 (b09zg9dd) [editorial]'
INFO: Trying 'hvfxsd1' mode: attempt 1 / 5
INFO: ffmpeg version string = 3.4.2-tessus
INFO: ffmpeg version number = 3.4
INFO: Creating subdirectory /Volumes/media/Downloads/TV Shows/Ordeal by Innocence for programme
INFO: File name prefix = Ordeal by Innocence.s01e02.Episode 2
INFO: Getting URL: http://b5rbsov.bidi.live.bbc.co.uk/vod-hls-uk-live/usp/auth/vod/piff_abr_full_hd/726a98-b0b0b6gz/vf_b0b0b6gz_7bb0e2d6-4be6-4b92-a31d-c3b0b685e513.ism.hlsv2.ism/vf_b0b0b6gz_7bb0e2d6-4be6-4b92-a31d-c3b0b685e513.ism.hlsv2-audio_eng=320000-video=1604000.m3u8?at=IS4uIEXgf166bc6796cdf1ffdb838d5db4866c449799025d5698665529b80
INFO: Begin downloading at: 0.00 MB (00:00:00) [1]
  1.0%     8.94 MB / ~893.77 MB (00:00:38 / 00:58:25) [    5 / 457] @  11.9 Mb/s ETA: 00:09:53 (hvfxsd1/bi) [audio+video]
  2.0%    17.88 MB / ~893.77 MB (00:01:17 / 00:58:25) [   10 / 457] @  13.0 Mb/s ETA: 00:08:58 (hvfxsd1/bi) [audio+video]
  3.0%    26.82 MB / ~893.77 MB (00:01:55 / 00:58:25) [   15 / 457] @  13.4 Mb/s ETA: 00:08:37 (hvfxsd1/bi) [audio+video]
  4.0%    35.76 MB / ~893.77 MB (00:02:26 / 00:58:25) [   19 / 457] @  12.4 Mb/s ETA: 00:09:11 (hvfxsd1/bi) [audio+video]
  5.0%    44.69 MB / ~893.77 MB (00:03:04 / 00:58:25) [   24 / 457] @  11.9 Mb/s ETA: 00:09:29 (hvfxsd1/bi) [audio+video]
  6.0%    53.63 MB / ~893.77 MB (00:03:43 / 00:58:25) [   29 / 457] @  11.9 Mb/s ETA: 00:09:23 (hvfxsd1/bi) [audio+video]
  7.0%    62.57 MB / ~893.77 MB (00:04:13 / 00:58:25) [   33 / 457] @  12.2 Mb/s ETA: 00:09:04 
==truncated==
 85.0%   759.78 MB / ~893.77 MB (00:55:41 / 00:58:25) [  435 / 457] @  10.8 Mb/s ETA: 00:01:38 (hvfxsd1/bi) [audio+video]
 86.0%   768.72 MB / ~893.77 MB (00:56:27 / 00:58:25) [  441 / 457] @  10.8 Mb/s ETA: 00:01:32 (hvfxsd1/bi) [audio+video]
 87.0%   777.66 MB / ~893.77 MB (00:57:05 / 00:58:25) [  446 / 457] @  10.9 Mb/s ETA: 00:01:25 (hvfxsd1/bi) [audio+video]
 88.0%   786.59 MB / ~893.77 MB (00:57:44 / 00:58:25) [  451 / 457] @  10.8 Mb/s ETA: 00:01:19 (hvfxsd1/bi) [audio+video]
INFO: Downloaded: 795.21 MB (00:58:25) [457] in 00:09:46 @ 10.86 Mb/s (hvfxsd1/bi) [audio+video]
INFO: Finished downloading to file: /Volumes/media/Downloads/TV Shows/Ordeal by Innocence/Ordeal by Innocence.s01e02.Episode 2.video.ts
INFO: Converting to MP4
INFO: Converting to MP4
INFO: Command: "/Applications/Get iPlayer Automator.app/Contents/MacOS/ffmpeg" "-loglevel" "fatal" "-stats" "-y" "-i" "/Volumes/media/Downloads/TV Shows/Ordeal by Innocence/Ordeal by Innocence.s01e02.Episode 2.hls.ts" "-c:v" "copy" "-c:a" "copy" "-bsf:a" "aac_adtstoasc" "/Volumes/media/Downloads/TV Shows/Ordeal by Innocence/Ordeal by Innocence.s01e02.Episode 2.partial.mp4"
INFO: Command exit code 0 (raw code = 0)
INFO: Loading recordings history
INFO: Downloading thumbnail
INFO: Getting URL: http://ichef.bbci.co.uk/images/ic/640x360/p06355yb.jpg
INFO: Tagging MP4
INFO: Command: "/Applications/Get iPlayer Automator.app/Contents/MacOS/AtomicParsley" "/Volumes/media/Downloads/TV Shows/Ordeal by Innocence/Ordeal by Innocence.s01e02.Episode 2.mp4" "--freefree" "--overWrite" "--stik" "TV Show" "--advisory" "remove" "--copyright" "2018 British Broadcasting Corporation, all rights reserved" "--title" "Episode 2" "--artist" "BBC One" "--albumArtist" "BBC TV" "--album" "Ordeal by Innocence: Series 1" "--grouping" "Drama,Crime" "--composer" "BBC iPlayer" "--genre" "Drama" "--comment" "Calgary is determined to prove that Jack died an innocent man." "--year" "2018-04-08T21:00:00+01:00" "--tracknum" "2" "--disk" "1" "--lyrics" "Calgary is determined to prove that Jack died an innocent man, but his alibi puts him in perilous danger of his own.
 Started writing to temp file.
INFO: Command exit code 0 (raw code = 0)
INFO: Getting URL: http://vod-sub-uk-live.bbcfmt.hs.llnwd.net/iplayer/subtitles/ng/modav/bUnknown-96cec08e-4fbe-4085-a1a3-56985f120dba_b0b0b6gz_1523227315948.xml?s=1523379457&e=1523422657&h=027417dcbf9a6733c25ebe2f686beb79
 Finished writing to temp file.
BBC Download (ID=b09zg9dd): Ordeal by Innocence: Series 1 - Episode 2 Completed Successfully
```

**Comment by skovatch** [comment]
It's easy enough to add the hvf modes to the list of modes that force HLS. I confess I didn't know that hvf is already Mac-friendly.

**Comment by skovatch** [comment]
Okay, adding 'hvfbest/better/good/etc.' and checking "Prefer 50 fps streams" will get this show to download in HD.

**Comment by sleaux-meaux** [comment]
All you need to do is select 50fps streams and Best as your BBC TV format. Even when it is fixed, the "force HLS" option doesn't really apply to this since get_iplayer prefers HLS to DASH by default.  But until it is fixed, you must deselect it.

**Comment by skovatch** [comment]
Fixed in 1.13.3
